stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/MessagesScreen.kt

```diff
@@ -224,169 +220,290 @@ public fun MessagesScreen(
+        )
+        AttachmentsPickerMenu(
+            attachmentsPickerViewModel = attachmentsPickerViewModel,
+            composerViewModel = composerViewModel
         )
+        MessageDialogs(listViewModel = listViewModel)
+    }
+}
 
-        var messageOptions by remember { mutableStateOf<List<MessageOptionItemState>>(emptyList()) }
+/**
+ * Contains the various menus and pickers the user
+ * can use to interact with messages.
+ *
+ * @param listViewModel The [MessageListViewModel] used to read state from.
+ * @param composerViewModel The [MessageComposerViewModel] used to read state from.
+ */
+@Composable
+private fun BoxScope.MessageMenus(
+    listViewModel: MessageListViewModel,
+    composerViewModel: MessageComposerViewModel,
+) {
+    val selectedMessageState = listViewModel.currentMessagesState.selectedMessageState
 
-        if (newMessageOptions.isNotEmpty()) {
-            messageOptions = newMessageOptions
-        }
+    val selectedMessage = selectedMessageState?.message ?: Message()
 
-        AnimatedVisibility(
-            visible = selectedMessageState is SelectedMessageOptionsState && selectedMessage.id.isNotEmpty(),
-            enter = fadeIn(),
-            exit = fadeOut(animationSpec = tween(durationMillis = AnimationConstants.DefaultDurationMillis / 2))
-        ) {
-            SelectedMessageMenu(
-                modifier = Modifier
-                    .align(Alignment.BottomCenter)
-                    .animateEnterExit(
-                        enter = slideInVertically(
-                            initialOffsetY = { height -> height },
-                            animationSpec = tween()
-                        ),
-                        exit = slideOutVertically(
-                            targetOffsetY = { height -> height },
-                            animationSpec = tween(durationMillis = AnimationConstants.DefaultDurationMillis / 2)
-                        )
+    MessagesScreenMenus(
+        listViewModel = listViewModel,
+        composerViewModel = composerViewModel,
+        selectedMessageState = selectedMessageState,
+        selectedMessage = selectedMessage
+    )
+
+    MessagesScreenReactionsPicker(
+        listViewModel = listViewModel,
+        composerViewModel = composerViewModel,
+        selectedMessageState = selectedMessageState,
+        selectedMessage = selectedMessage,
+    )
+}
+
+/**
+ * Contains selected message and reactions menus
+ * wrapped inside an animated composable.
+ *
+ * @param listViewModel The [MessageListViewModel] used to read state and
+ * perform actions.
+ * @param composerViewModel The [MessageComposerViewModel] used to read state and
+ * perform actions.
+ * @param selectedMessageState The state of the currently selected message.
+ * @param selectedMessage The currently selected message.
+ */
+@Suppress("LongMethod")
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+private fun BoxScope.MessagesScreenMenus(
```

Comment:
If you have a better naming suggestion I would love to hear it.

These are not necessarily grouped by function, but by having them recompose as little as possible and without duplicating calculations and outputs.

Reply:
I think this is fine, it's clear and respects the functionality that it shows in the UI

Reply:
I would name it SelectedMessageOptions maybe? All of the functionality in it is tied to the selected message, i think it better describes the functionality and will follow in line with `MessagesScreenReactionsPicker` and `AttachmentsPickerMenu` whose naming reflects their function.